tests/v2/validation/observability/rbac/stackstate_rbac_test.go

```diff
@@ -0,0 +1,257 @@
+	require.NoError(rb.T(), err, "Unable to create stackstate namespace.")
+
+	log.Info("Verifying if the ui plugin repository for ui extensions exists.")
+	_, err = rb.client.Catalog.ClusterRepos().Get(context.TODO(), rancherUIPlugins, meta.GetOptions{})
+
+	if k8sErrors.IsNotFound(err) {
+		err = observability.AddExtensionsRepo(rb.client, rancherUIPlugins, uiExtensionsRepo, uiGitBranch)
+		require.NoError(rb.T(), err)
+	} else {
+		require.NoError(rb.T(), err)
+	}
+
+	var stackstateConfigs observability.StackStateConfigs
+	config.LoadConfig(stackStateConfigFileKey, &stackstateConfigs)
+	rb.stackstateConfigs = stackstateConfigs
+
+	log.Info("Crete a node driver with stackstate extensions ui to whitelist stackstate URL")
+	_, err = client.Management.NodeDriver.ByID(observability.StackstateName)
+	if strings.Contains(err.Error(), "Not Found") {
+		err = observability.InstallNodeDriver(rb.client, []string{rb.stackstateConfigs.Url})
+		require.NoError(rb.T(), err, "Unable to install node driver for stackstate.")
+	} else {
+		require.NoError(rb.T(), err)
+	}
+
+	rb.T().Log("Checking if the stack state CRD is installed.")
+	crdsExists, err := rb.client.Steve.SteveType(observability.ApiExtenisonsCRD).ByID(observability.ObservabilitySteveType)
+	if crdsExists == nil && strings.Contains(err.Error(), "Not Found") {
+		err = observability.InstallStackstateCRD(rb.client)
+		require.NoError(rb.T(), err, "Unable to install stackstate CRD.")
+	} else {
+		require.NoError(rb.T(), err)
+	}
+
+	client, err = client.ReLogin()
+	require.NoError(rb.T(), err)
+
+	rb.T().Log("Checking if the stack state extension is already installed.")
+	initialStackstateExtension, err := extencharts.GetChartStatus(client, localCluster, charts.StackstateExtensionNamespace, charts.StackstateExtensionsName)
+	require.NoError(rb.T(), err)
+
+	if !initialStackstateExtension.IsAlreadyInstalled {
+		latestUIPluginVersion, err := rb.client.Catalog.GetLatestChartVersion(charts.StackstateExtensionsName, charts.UIPluginName)
+		require.NoError(rb.T(), err)
+
+		rb.T().Log("Installing stackstate ui extensions")
+		extensionOptions := &charts.ExtensionOptions{
+			ChartName:   charts.StackstateExtensionsName,
+			ReleaseName: charts.StackstateExtensionsName,
+			Version:     latestUIPluginVersion,
+		}
+
+		err = charts.InstallStackstateExtension(client, extensionOptions)
+		require.NoError(rb.T(), err)
+
+		log.Info("Adding stack state extension configuration.")
+
+		steveAdminClient, err := client.Steve.ProxyDownstream(localCluster)
+		require.NoError(rb.T(), err)
+
+		crdConfig := observability.NewStackstateCRDConfiguration(charts.StackstateNamespace, rb.stackstateConfigs)
+		crd, err := steveAdminClient.SteveType(charts.StackstateCRD).Create(crdConfig)
+		require.NoError(rb.T(), err, "Unable to install stackstate CRD configuration.")
+
+		_, err = steveAdminClient.SteveType(charts.StackstateCRD).ByID(crd.ID)
```

Comment:
if this is meant to be a sort of 'verify the chart was installed' I would add an explicit func for that instead of assuming that if the CRD is there, it is correctly installed. Otherwise, this line is redundant afaik because the previous requireNoError would have thrown an error if the crd.ID wasn't created.